dotCMS/src/main/java/com/dotmarketing/portlets/workflows/actionlet/PushNowActionlet.java

```diff
@@ -73,15 +72,21 @@ public void executeAction(WorkflowProcessor processor, Map<String, WorkflowActio
             for (String name : whereToSend) {
                 if (UtilMethods.isSet(name)) {
                     name = name.trim();
-                    Environment e = APILocator.getEnvironmentAPI().findEnvironmentByName(name);
+                    final Environment e = APILocator.getEnvironmentAPI().findEnvironmentByName(name);
                     if (e != null) {
 
                         envsToSendTo.add(e);
+                    }else{
+                        Logger.error(PushNowActionlet.class, "The Environment " + name + " does not exists");
                     }
                 }
 
             }
 
+            if(envsToSendTo.isEmpty()){
+                throw new DotPublisherException("There are no enviroments to send the bundle");
```

Comment:
Can you add the Contentlet's Identifier here please? It'd really help for support. Maybe update the error message to something like this:
```
"There are no enviroments to send the bundle containing Contentlet " + ref.getIdentifier()
```

Reply:
agree